podvm: Failed to build `fedora-binaries-builder` on s390x

### Describe the bug

After https://github.com/confidential-containers/cloud-api-adaptor/pull/2026 gets merged,  building `fedora-binaries-builder` fails on s390x.


### How to reproduce

Run the following commands on an s390x machine.

```
cd src/cloud-api-adaptor/podvm-mkosi
make fedora-binaries-builder
```

### CoCo version information

any

### What TEE are you seeing the problem on

None

### Failing command and relevant log output

```shell
=> [12/14] ADD https://github.com/oras-project/oras/releases/download/v1.2.0/oras_1.2.0_linux_s390x.tar.gz oras_1.2.0_linux_amd64.tar.gz                                        0.0s
 => ERROR [13/14] RUN rm -rf /usr/local/bin/oras && tar -C /usr/local/bin -xzf oras_1.2.0_linux_s390x.tar.gz && rm -f oras_1.2.0_linux_s390x.tar.gz                              0.6s
------
 > [13/14] RUN rm -rf /usr/local/bin/oras && tar -C /usr/local/bin -xzf oras_1.2.0_linux_s390x.tar.gz && rm -f oras_1.2.0_linux_s390x.tar.gz:
0.588 tar (child): oras_1.2.0_linux_s390x.tar.gz: Cannot open: No such file or directory
0.588 tar (child): Error is not recoverable: exiting now
0.588 tar: Child returned status 2
0.588 tar: Error is not recoverable: exiting now
------
Dockerfile.podvm_builder.fedora:46
--------------------
  44 |
  45 |     ADD https://github.com/oras-project/oras/releases/download/v${ORAS_VERSION}/oras_${ORAS_VERSION}_linux_${ARCH}.tar.gz oras_${ORAS_VERSION}_linux_amd64.tar.gz
  46 | >>> RUN rm -rf /usr/local/bin/oras && tar -C /usr/local/bin -xzf oras_${ORAS_VERSION}_linux_${ARCH}.tar.gz && rm -f oras_${ORAS_VERSION}_linux_${ARCH}.tar.gz
  47 |
  48 |     WORKDIR /src
--------------------
ERROR: failed to solve: process "/bin/sh -c rm -rf /usr/local/bin/oras && tar -C /usr/local/bin -xzf oras_${ORAS_VERSION}_linux_${ARCH}.tar.gz && rm -f oras_${ORAS_VERSION}_linux_${ARCH}.tar.gz" did not complete successfully: exit code: 2
make: *** [Makefile:32: fedora-binaries-builder] Error 1
```

--- a/src/cloud-api-adaptor/podvm/Dockerfile.podvm_builder.fedora
+++ b/src/cloud-api-adaptor/podvm/Dockerfile.podvm_builder.fedora
@@ -24,7 +24,7 @@ RUN dnf groupinstall -y 'Development Tools' && \
     dnf install -y yum-utils gnupg git perl-core pkg-config libseccomp-devel gpgme-devel \
     device-mapper-devel unzip libassuan-devel \
     perl-FindBin openssl-devel tpm2-tss-devel \
-    clang which && \
+    clang which xz && \
     dnf clean all
 
 ADD https://dl.google.com/go/go${GO_VERSION}.linux-${ARCH}.tar.gz go${GO_VERSION}.linux-${ARCH}.tar.gz
@@ -42,7 +42,7 @@ RUN chmod a+x rustup && ./rustup -y --default-toolchain ${RUST_VERSION}
 ADD https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-${PROTOC_ARCH}.zip protoc-${PROTOC_VERSION}-linux-${PROTOC_ARCH}.zip
 RUN unzip protoc-${PROTOC_VERSION}-linux-${PROTOC_ARCH}.zip -d /usr/local && rm -f protoc-${PROTOC_VERSION}-linux-${PROTOC_ARCH}.zip
 
-ADD https://github.com/oras-project/oras/releases/download/v${ORAS_VERSION}/oras_${ORAS_VERSION}_linux_${ARCH}.tar.gz oras_${ORAS_VERSION}_linux_amd64.tar.gz
+ADD https://github.com/oras-project/oras/releases/download/v${ORAS_VERSION}/oras_${ORAS_VERSION}_linux_${ARCH}.tar.gz oras_${ORAS_VERSION}_linux_${ARCH}.tar.gz
 RUN rm -rf /usr/local/bin/oras && tar -C /usr/local/bin -xzf oras_${ORAS_VERSION}_linux_${ARCH}.tar.gz && rm -f oras_${ORAS_VERSION}_linux_${ARCH}.tar.gz
 
 WORKDIR /src

